jetty-ee9/jetty-ee9-servlet/src/test/java/org/eclipse/jetty/ee9/servlet/DefaultServletTest.java

```diff
@@ -138,18 +134,29 @@ public void init(WorkDir workDir) throws Exception
     @AfterEach
     public void destroy() throws Exception
     {
-        server.stop();
-        server.join();
+        if (server != null)
+        {
+            server.stop();
+            server.join();
+        }
         assertThat(FileSystemPool.INSTANCE.mounts(), empty());
     }
 
     @Test
     public void testListingWithSession() throws Exception
     {
-        ServletHolder defholder = context.addServlet(DefaultServlet.class, "/*");
-        defholder.setInitParameter("dirAllowed", "true");
-        defholder.setInitParameter("redirectWelcome", "false");
-        defholder.setInitParameter("gzip", "false");
+        Path docRoot = workDir.getEmptyPathDir().resolve("docroot");
+        FS.ensureDirExists(docRoot);
+
+        startServer((context) ->
+        {
+            context.setBaseResource(ResourceFactory.of(context).newResource(docRoot));
+
+            ServletHolder defholder = context.addServlet(DefaultServlet.class, "/*");
```

Comment:
I know it's this way in the original file, however the `DefaultServlet` is normally mapped at `/`, not `/*`

Reply:
Looking back historically, this seems intentional.
This servlet isn't declared with the "default" name either.
It seems to want to replicate a end user use-case.

Keep in mind that the new test cases coming after this PR is merged will introduce even more horrible DefaultServlet usages.

Reply:
OK, maybe a javadoc comment then to ensure that in future we know this is intentional for the test.